### PR TITLE
Buildkite dynamic pipelines

### PR DESCRIPTION
**Pull request by edsavage**
Use dynamically generated `BuildKite` pipelines to:

- Promote code re-use between pipelines
- Customize build based on PR labels and/or comments

### CODE REVIEW

**Comment by edsavage** [comment]
buildkite debug on linux

**Review by droberts195 — COMMENTED** [review]
One thing I am not completely clear on is when we do native builds with testing on `linux-aarch64`, and when we do cross compiles (which can't run unit tests). Which variations are we running in the different types of builds?

**Comment by edsavage** [comment]
> One thing I am not completely clear on is when we do native builds with testing on `linux-aarch64`, and when we do cross compiles (which can't run unit tests). Which variations are we running in the different types of builds?

Currently, we do native builds with testing for both PR builds and branch builds. Cross compilation of `linux-aarch64` is also triggered for both PR and branch builds, but only for non-debug builds. 

As discussed, I'll change the logic so that we always perform a cross compilation with debug enabled on every PR build - even if the rest of the build is optimised. Cross compilation of `linux-aarch64` will not be performed under any other circumstances.

File: .buildkite/ml_pipeline/config.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env Python
```
Comment (droberts195) [inline .buildkite/ml_pipeline/config.py:1]: Should it be lower case?

```suggestion
#!/usr/bin/env python
```

**Review by droberts195 — APPROVED** [review]
LGTM

The logic is so complex that I am sure this will need another iteration to get the branch builds working perfectly rather than just the PR builds. But we have to merge it to do that, so I'm happy to merge and see what happens.

**Pull request merged**